exporter/datadogexporter/translate_traces.go

```diff
@@ -0,0 +1,517 @@
+		spans := ils.Spans()
+		for j := 0; j < spans.Len(); j++ {
+			span, err := spanToDatadogSpan(spans.At(j), localServiceName, datadogTags)
+
+			if err != nil {
+				return payload, err
+			} else {
+				var apiTrace *pb.APITrace
+				var ok bool
+
+				if apiTrace, ok = apiTraces[span.TraceID]; !ok {
+					apiTrace = &pb.APITrace{
+						TraceID:   span.TraceID,
+						Spans:     []*pb.Span{},
+						StartTime: 0,
+						EndTime:   0,
+					}
+					apiTraces[apiTrace.TraceID] = apiTrace
+				}
+				
+				addToAPITrace(apiTrace, span)
+			}
+		}
+	}
+
+	for _, apiTrace := range apiTraces {
+		top := apm.GetAnalyzedSpans(apiTrace.Spans)
+		apm.ComputeSublayerMetrics(apiTrace.Spans)
+		payload.Transactions = append(payload.Transactions, top...)
+		payload.Traces = append(payload.Traces, apiTrace)
+	}
+
+	return payload, nil
+}
+
+
+// convertSpan takes an internal span representation and returns a Datadog span.
+func spanToDatadogSpan(	s pdata.Span,
+	localServiceName string,
+	datadogTags map[string]string,
+) (*pb.Span, error) {
+	tags := aggregateSpanTags(s, datadogTags)
+
+	// get tracestate as just a general tag
+	if len(s.TraceState()) > 0 {
+		tags[tracetranslator.TagW3CTraceState] = string(s.TraceState())
+	}
+
+	// get start/end time to calc duration
+	startTime := s.StartTime()
+	endTime := s.EndTime()
+	duration := int64(endTime) - int64(startTime)
+
+	// it's possiblle end time is unset, so default to 0 rather than using a negative number
+	if s.EndTime() == 0 {
+		duration = int64(0)
+	}
+
+	datadogType := spanKindToDatadogType(s.Kind())
+
+	span := &pb.Span{
+		TraceID:  decodeAPMId(hex.EncodeToString(s.TraceID().Bytes()[:])),
+		SpanID:  decodeAPMId(hex.EncodeToString(s.SpanID().Bytes()[:])),
+		Name:     getDatadogSpanName(s, tags),
+		Resource: getDatadogResourceName(s, tags),
+		Service:  "service_example",
+		Start:    int64(startTime),
+		Duration: int64(duration),
+		Metrics:  map[string]float64{},
+		Meta:     map[string]string{},
+		Type: datadogType,
+	}
+
+	// TODO: confirm parentId approach
+	if len(s.ParentSpanID().Bytes()) > 0 {
+		idVal := decodeAPMId(hex.EncodeToString(s.ParentSpanID().Bytes()[:]))
+
+		span.ParentID = idVal
+	} else {
+		span.ParentID = 0
+	}
+
+	// Set Span Status and any response or error details
+	status := s.Status()
+	if !status.IsNil() {
+		// map to known msg + status
+		code, ok := statusCodes[int32(status.Code())]
+		if !ok {
+			code = codeDetails{
+				message: "ERR_CODE_" + strconv.FormatInt(int64(status.Code()), 10),
+				status:  http.StatusInternalServerError,
+			}
+		}
+
+		// check if 500 or 400 level, if so set error
+		if code.status/100 == 5 || code.status/100 == 4 {
+			span.Error = 1
+			// set error type
+			tags[ext.ErrorType] = code.message
+			// set error message
+			if status.Message() != "" {
+				tags[ext.ErrorMsg] = status.Message()
+			} else {
+				tags[ext.ErrorMsg] = code.message
+			}
+		// otherwise no error
+		} else {
+			span.Error = 0
+		}
+
```

Comment:
nit: we can simplify this since `span.Error` is set to 0 by default (or leave it /a comment for emphasis)
```suggestion
		}
```